From c3685018d57ede69213fc2b3c8c6ce001a46b4fa Mon Sep 17 00:00:00 2001
From: Adrien Ricciardi <aricciardi@baylibre.com>
Date: Tue, 17 Oct 2023 17:26:43 +0200
Subject: [PATCH 35/38] examples: posix: Used the new tpl_can_fill_pdu_info()
 function in the CAN demo to simplify the creation of the CAN frames to send.

Signed-off-by: Adrien Ricciardi <aricciardi@baylibre.com>
---
 examples/posix/can_demo/can_demo.c | 12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)

diff --git a/examples/posix/can_demo/can_demo.c b/examples/posix/can_demo/can_demo.c
index fb1eb138..efcdee47 100644
--- a/examples/posix/can_demo/can_demo.c
+++ b/examples/posix/can_demo/can_demo.c
@@ -100,12 +100,8 @@ TASK(can_task)
 	int i;
 
 	printf("Transmitting a CAN 2.0 frame with standard ID...\r\n");
-	can_pdu.id = 0x123 | TPL_CAN_ID_TYPE_STANDARD;
 	strcpy((char *) payload, "Ciao !");
-	can_pdu.length = strlen((char *) payload);
-	can_pdu.sdu = payload;
-	pdu_info.SduDataPtr = (uint8 *) &can_pdu;
-	pdu_info.SduLength = sizeof(can_pdu);
+	tpl_can_fill_pdu_info(&can_pdu, &pdu_info, 0x123 | TPL_CAN_ID_TYPE_STANDARD, payload, strlen((char *) payload));
 	ret = CanIf_Transmit(0, &pdu_info);
 	if (ret)
 		printf("[%s:%d] Error : failed to transmit the frame (%d).\r\n", __func__, __LINE__, ret);
@@ -130,12 +126,8 @@ TASK(can_task)
 	}
 
 	printf("Transmitting a CAN-FD frame with extended ID...\r\n");
-	can_pdu.id = 0xABCD123 | TPL_CAN_ID_TYPE_FD_EXTENDED;
 	strcpy((char *) payload, "This is a longer string.");
-	can_pdu.length = strlen((char *) payload); // The FD length will be automatically adapted by the driver
-	can_pdu.sdu = payload;
-	pdu_info.SduDataPtr = (uint8 *) &can_pdu;
-	pdu_info.SduLength = sizeof(can_pdu);
+	tpl_can_fill_pdu_info(&can_pdu, &pdu_info, 0xABCD123 | TPL_CAN_ID_TYPE_FD_EXTENDED, payload, strlen((char *) payload)); // The FD length will be automatically adapted by the driver
 	ret = CanIf_Transmit(1, &pdu_info);
 	if (ret)
 		printf("[%s:%d] Error : failed to transmit the frame (%d).\r\n", __func__, __LINE__, ret);
-- 
2.34.1

